Revamp Developer Portal: Add Supported Authentication Types reference page (#1960)
TykTechnologies/tyk-docs: portal/portal-auth-types.mdx

---
title: "Developer Portal Authentication Support"
description: "Reference for authentication types in the Tyk Developer Portal: credential lifecycle, consumer actions, admin actions, and combination rules."
sidebarTitle: "API Authentication"
keywords: "Developer Portal, authentication types, API key, OAuth 2.0, DCR, JWT, mTLS, custom credentials, credential lifecycle, self-serve, managed"
---

> When you add APIs to an API Product, the Developer Portal enforces that all APIs in the product use an authentication method. This page summarises which authentication types the Portal supports, what consumers and admins can do with each, and which types can be combined.

Reviewer: ```suggestion
When an API Consumer is granted access to an API Product they will be issued a credential (for example an Auth Token) to present in the API request. For this reason, an API Product can only contain APIs that use a common [authentication method](/portal/api-products#authentication-method).

This page summarises which methods the Portal supports, what API Consumers and API Owners can do with each, and which types can be combined.
```

<Note>

> This page covers authentication as managed by the Developer Portal. For Tyk Gateway-level authentication configuration, see [API Authentication](/api-management/authentication).

Reviewer: ```suggestion
This page covers authentication methods from the perspective of the Developer Portal. For configuration of client authentication at the Tyk Gateway-level, see [API Authentication](/api-management/client-authentication).
```

</Note>

> ## Credential Ownership: Self-Serve vs Managed

Reviewer: ```suggestion
## Credential Management: Self-Serve vs Managed
```

> Before reviewing the table, it helps to understand the two credential ownership modes the Portal uses:

Reviewer: ```suggestion
The Portal supports two methods for credential issuance and management:
```

> **Self-Serve:** The API Consumer creates, rotates, and revokes their own credentials from the Live Portal without Admin involvement.

Reviewer: ```suggestion
**Self-Serve:** The API Consumer creates, rotates, and revokes their own credentials from the Live Portal.
```

> **Managed:** The Portal Admin imports or pre-configures credentials. Consumers have view-only access, or can revoke but not create or rotate credentials.

Reviewer: ```suggestion
**Managed:** The API Owner manages the creation and rotation of credentials. API Consumers have view-only access, or can revoke but not create or rotate credentials.
```

The **Mode** column in the table below indicates which model applies to each authentication type.

## Supported Authentication Types

> | Auth Type | Supported | Mode | Consumer Actions | Admin Actions |

Author: Consumer Actions and Admin action columns can be removed

Reviewer: I'm not sure there's any value from the "supported" column, as this only lists "supported auth types".

|---|---|---|---|---|

> | [**API Token / API Key**](/api-management/authentication/bearer-token) | Yes | Self-Serve | Request access, rotate credential, revoke credential | View, edit, and revoke credentials |

Reviewer: ```suggestion
| [**Auth Token / API Key**](/api-management/authentication/bearer-token) | Self-Serve |
```

We use the term "Auth Token" in the docs and Dashboard; Portal UI uses API Key (terms will be aligned in a future UI update)

> | [**OAuth 2.0 (via DCR)**](/api-management/authentication/oauth-2) | Yes | Self-Serve | Request access (generates Client ID and Secret), rotate secret, revoke credential | Configure DCR template, view credentials, revoke credentials |

Reviewer: ```suggestion
```

There is no "OAuth 2.0" option at this time.

> | [**JWT**](/basic-config-and-security/security/authentication-authorization/json-web-tokens) | Yes | Self-Serve | Request access, rotate credential, revoke credential | Configure DCR template, view credentials, revoke credentials |

Reviewer: ```suggestion
| [**JWT / OAuth**](/basic-config-and-security/security/authentication-authorization/json-web-tokens) | Self-Serve |
```

We recommend that JWT Auth is used for OAuth 2.0 secured APIs (unless using the Tyk as auth server (Tyk OAuth) method).

> | [**mTLS (Self-Serve)**](/api-management/authentication/certificate-auth) | Yes | Self-Serve | Upload client certificate during access request, revoke access | View credentials, revoke access |

Reviewer: ```suggestion
```

The "standalone" mTLS option doesn't actually do anything - while it's conceptually possible to create an API that would appear with this option, it wouldn't actually work at runtime. Let's leave this off - aiming to fix the UX here in a later release.

> | [**mTLS (Managed)**](/api-management/authentication/certificate-auth) | Yes | Managed | Request access using organization certificates, view credentials, revoke access | Pre-configure and manage certificates per organization in the Admin Portal |

Reviewer: ```suggestion
```

> | [**Auth Token + mTLS**](/api-management/authentication/certificate-auth) | Yes | Self-Serve or Managed | Request access; issued credential is automatically bound to the consumer's certificate | Enable certificate-token binding on the Product |

Reviewer: ```suggestion
| [**Auth Token + mTLS**](/api-management/implement-tls#using-a-static-client-certificate-allow-list) | Self-Serve or Managed |
```

> | [**JWT + mTLS**](/api-management/authentication/certificate-auth) | Yes | Self-Serve or Managed | Request access (requires DCR configuration at checkout), upload client certificate (if self-serve), rotate secret, revoke credential | Configure DCR template, view credentials, revoke credentials, manage certificates (if managed) |

Reviewer: ```suggestion
| [**JWT + mTLS**](/api-management/implement-tls#using-a-static-client-certificate-allow-list) | Self-Serve or Managed |
```

| [**Basic Authentication**](/api-management/authentication/basic-authentication) | Yes | Managed | View credential details (read-only); revoke credential | Create, edit, and assign credentials via Admin Portal or Portal API |
| [**Custom Authentication**](/api-management/authentication/custom-auth) | Yes | Managed | View credential details (read-only); revoke credential | Create, edit, and assign credentials via Admin Portal or Portal API |

> | [**Custom Credentials (Feature)**](/tyk-developer-portal/import-custom-credentials) | Yes | Managed | View credential details (read-only); revoke credential; for Tyk-managed custom credentials: rotate credential | Create, edit, and assign credentials via Admin Portal or Portal API |

Reviewer: ```suggestion
```

This isn't an "auth type" -> I believe it is just Auth Token, where the credentials are imported to Tyk.

> **Unsupported authentication types:** [HMAC](/basic-config-and-security/security/authentication-authorization/hmac-signatures), [OpenID Connect (OIDC)](/api-management/single-sign-on-oidc), and [Keyless](/basic-config-and-security/security/authentication-authorization/open-keyless) are not supported by the Portal.

Reviewer: ```suggestion
### Unsupported authentication types

APIs secured with [HMAC](/basic-config-and-security/security/authentication-authorization/hmac-signatures), [Tyk OAuth 2.0](/api-management/authentication/oauth-2), [Certificate Auth](/api-management/authentication/certificate-auth) or the deprecated [OpenID Connect (OIDC)](/api-management/client-authentication#integrate-with-openid-connect-deprecated) and [External OAuth](/api-management/client-authentication#integrate-with-external-authorization-server-deprecated) methods are not supported by the Portal and cannot be included in API Products.

[Open](/basic-config-and-security/security/authentication-authorization/open-keyless) (unsecured or keyless) APIs with no client authentication are also not supported.
```

> ## OAuth 2.0 and DCR

Reviewer: ```suggestion
## Dynamic Client Registration (DCR) for OAuth 2.0
```

> All OAuth 2.0 authentication in the Developer Portal uses [Dynamic Client Registration (DCR)](/tyk-stack/tyk-developer-portal/enterprise-developer-portal/api-access/dynamic-client-registration). The Tyk internal OAuth provider (`oauth` auth type) is not supported.

Reviewer: ```suggestion
When an API Product grants access to APIs secured using JSON Web Tokens (JWT), the tokens are issued by external Identity Providers (IdPs).

When an API Consumer requests access for their developer app, the Portal uses [Dynamic Client Registration (DCR)](/tyk-stack/tyk-developer-portal/enterprise-developer-portal/api-access/dynamic-client-registration) to automatically register an OAuth 2.0 client with the IdP on their behalf, returning a `client_id` and `client_secret` to the developer. The API Consumer then uses those credentials to obtain a JWT from the IdP, which they present to the Tyk Gateway. Tyk validates the JWT and creates a [Session](/api-management/access-control/overview#2-dynamically-generated-sessions) dynamically — no pre-registration with Tyk is required.

This differs from the [Authentication Token](/api-management/authentication/bearer-token) method, where Tyk issues the access token directly.

<Note>
The Tyk OAuth 2.0 authentication method, which uses Tyk as the auth server is not supported by the Developer Portal.
</Note>
```

### Supported Identity Providers

| Identity Provider | Support | Notes |
|---|---|---|
| Keycloak | Fully supported | N/A |
| Okta | Fully supported | Requires the `OktaAppType` field (`web`, `native`, `browser`, or `service`). For Client Credentials flow, set the allowed response type to `token` only. |
| Gluu | Fully supported | N/A |
| Curity | Supported via Other type | Curity requires authentication on its DCR endpoint by default. Workaround: disable DCR endpoint authentication in Curity (no-authentication mode). |
| Any RFC-compliant IdP | Supported via Other type | Select **Other** in the DCR template settings. Full compatibility is not guaranteed. |

## Interactive Documentation and Playground Authentication

> The Portal automatically handles authentication when consumers use interactive API documentation or testing interfaces.

Reviewer: ```suggestion
The Portal automatically handles authentication when Consumers use interactive API documentation or testing interfaces.
```

> **Automatic credential injection:** When an authenticated consumer opens API documentation (Stoplight, ReDoc, or AsyncAPI) or the GraphQL Playground, the Portal automatically populates their approved credentials for that product. Consumers can test APIs directly from the documentation without manually copying tokens or headers.

Author: @andyo-tyk I am not sure about this one? I have never seen it.

Reviewer: ```suggestion
```

Agreed, this doesn't seem to be a supported functionality

## Combination Rules

All APIs in an API Product must share a compatible authentication method. The Portal issues a single credential that must work across every API in the product.

> ### Supported combinations

Reviewer: I'm not sure we need this section -> we've already stated what the options are above.
